src/js/render_wordle.js

```diff
@@ -116,6 +118,89 @@ inverted_wordles.makeLayout = function (instance) {
     return layout;
 };
 
+/** Return the computed value of the font size of the given DOM element.
+ * @param {Object} elm - A DOM element
+ * @return {Number} The computed value of the font size of the given element.
+ */
+inverted_wordles.getFontSizeValue = function (elm) {
+    const style = window.getComputedStyle(elm, null).getPropertyValue("font-size");
+    return parseFloat(style);
+};
+
+/** Return the number at the midpoint if length is odd, otherwise the average of the two middle numbers.
+ * See: https://www.w3resource.com/javascript-exercises/fundamental/javascript-fundamental-exercise-88.php
+ * @param {Array} numArray - An array of numbers
+ * @return {Number} The median value of values in an array
+ */
+inverted_wordles.getMedianFontSize = function (numArray) {
+    const mid = Math.floor(numArray.length / 2), nums = [...numArray].sort((a, b) => a - b);
+    return numArray.length % 2 !== 0 ? nums[mid] : (nums[mid - 1] + nums[mid]) / 2;
+};
+
+/** Finds font sizes of all wordle texts and calculate these numbers:
+ * 1. The media number of all font sizes;
+ * 2. The gap number between min and median font sizes;
+ * 3. The gap number between media and max font sizes.
+ * @param {Object} instance - The singleton instance. Calculated numbers are returned by attaching to it.
+ */
+inverted_wordles.calSizeGaps = function (instance) {
+    let fontSizes = [];
+    document.querySelectorAll("text.wordle-text").forEach(elm => {
+        fontSizes.push(inverted_wordles.getFontSizeValue(elm));
+    });
+
+    const minFontSize = Math.min(...fontSizes);
+    const maxFontSize = Math.max(...fontSizes);
+
+    instance.medianFontSize = inverted_wordles.getMedianFontSize(fontSizes);
+    instance.minMedianFontSizeGap = instance.medianFontSize - minFontSize;
+    instance.maxMedianFontSizeGap = maxFontSize - instance.medianFontSize;
+};
+
+/** Use web speech API to read wordle texts. The pitch of the voice represents the font size of the text.
+ * The larger the font size, the higer should be the pitch.
+ *
+ * The algorithm that calculates the pitch value based on the wordle font size:
+ * Note: the pitch value accepted by web speech API is a number in a range of 0 to 2. 1 is the default value.
+ * See: https://developer.mozilla.org/en-US/docs/Web/API/SpeechSynthesisUtterance/pitch
+ * So, the pitch is calculated based on the default value.
+ * 1. Find all font sizes of wordle texts;
+ * 2. If the text font size === the median font size, pitch = 1;
+ * 3. If the text font size >= the median font size, pitch = 1 + (fontSize - medianSize) / maxSize - medianSize;
+ * 2. If the text font size < the median font size, pitch = 1 + (fontSize - medianSize) / medianSize - minSize;
+ * @param {Object} instance - The calculated numbers are returned by attaching to the `instance` object
+ */
+inverted_wordles.bindTTS = function (instance) {
+    inverted_wordles.calSizeGaps(instance);
+
+    document.querySelectorAll("text.wordle-text").forEach(elm => {
```

Comment:
Most of this implementation is in the wrong place - it should instead be attached to "makeLayout" so that it is responsive to changes in the layout caused by an update in the words or change of strategy. Only the "pointerover" event binding should be in this function, the pitch computation should go into a different function.